components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineView.kt

```diff
@@ -85,10 +85,19 @@ class GeckoEngineView @JvmOverloads constructor(
         }
     }
 
+    @Synchronized
+    override fun release() {
+        currentSession?.apply { unregister(observer) }
+
+        currentSession = null
+
+        currentGeckoView.releaseSession()
+    }
+
     override fun onDetachedFromWindow() {
         super.onDetachedFromWindow()
 
-        currentSession?.apply { unregister(observer) }
+        release()
```

Comment:
I wonder why this change is on nightly but not beta 🤔 ? As beta and nightly have the same behavior for for the`release` function.

Reply:
Oh, yeah, that's something I need to port over to beta.

Reply:
Updated!